client-akka-http/src/main/scala/bekindrewind/akkahttp/VcrHttp.scala

```diff
@@ -0,0 +1,129 @@
+package bekindrewind.akkahttp
+
+import bekindrewind._
+
+import akka.actor.ClassicActorSystemProvider
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.util.ByteString
+import io.circe.parser._
+import io.circe.syntax._
+
+import java.io.Closeable
+import java.net.URI
+import java.nio.file.Path
+import java.time.OffsetDateTime
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Try
+
+object VcrHttp {
+  def create(
+    sendRequest: HttpRequest => Future[HttpResponse],
+    recordingPath: Path,
+    recordOptions: RecordOptions = RecordOptions.default,
+    matcher: VcrMatcher = VcrMatcher.groupBy(r => (r.method, r.uri))
+  )(implicit executionContext: ExecutionContext, materializer: Materializer): VcrHttp =
+    new VcrHttp(sendRequest, recordingPath, recordOptions, matcher)
+
+  def useClassicActorSystem(
+    recordingPath: Path,
+    recordOptions: RecordOptions = RecordOptions.default,
+    matcher: VcrMatcher = VcrMatcher.groupBy(r => (r.method, r.uri)),
+    executionContext: Option[ExecutionContext] = None
+  )(implicit system: ClassicActorSystemProvider): VcrHttp = {
+    implicit val ec = executionContext.getOrElse(system.classicSystem.dispatcher)
+    create(Http().singleRequest(_), recordingPath, recordOptions, matcher)
+  }
+
+  private[akkahttp] def toVcrRequest(request: HttpRequest): VcrRecordRequest =
+    VcrRecordRequest(
+      method = request.method.value,
+      uri = URI.create(request.uri.toString()),
+      body = request.entity.toString,
```

Comment:
I think you may have to pattern match this and convert to String in each case (unless akka-http has a helper that does that for you). Otherwise you'll get something like `"body": "HttpEntity.Strict(none/none,0 bytes total)"` in the JSON.

Reply:
Oops.
Entity should be passed to `unmarshal`.
Now fixed.